Spec/RealtimeClientPresence.swift

```diff
@@ -316,6 +380,426 @@ class RealtimeClientPresence: QuickSpec {
+
+                            let client = AblyTests.newRealtime(options)
+                            defer { client.dispose(); client.close() }
+                            let channel = client.channels.get("foo")
+                            waitUntil(timeout: testTimeout) { done in
+                                client.connection.once(.Connected) { stateChange in
+                                    expect(stateChange?.reason).to(beNil())
+                                    done()
+                                }
+                            }
+
+                            guard let connectionId = client.connection.id else {
+                                fail("Should have a connection ID"); return
+                            }
+                            guard let transport = client.transport as? TestProxyTransport else {
+                                fail("TestProxyTransport is not set"); return
+                            }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                let partialDone = AblyTests.splitDone(4, done: done)
+                                let localMember = ARTPresenceMessage(clientId: "local1", action: .Enter, connectionId: connectionId, id: "\(connectionId):1:1", timestamp: NSDate())
+
+                                channel.once(.Attaching) { stateChange in
+                                    // Local member
+                                    channel.presenceMap.put(localMember)
+                                    partialDone()
+                                }
+                                channel.attach() { error in
+                                    expect(error).to(beNil())
+                                    partialDone()
+                                }
+
+                                transport.beforeProcessingReceivedMessage = { protocolMessage in
+                                    if protocolMessage.action == .Attached {
+                                        // Expect a Sync
+                                        expect(protocolMessage.hasPresence).to(beTrue())
+                                        expect(protocolMessage.resumed).to(beFalse())
+                                    }
+                                    else if protocolMessage.action == .Sync {
+                                        transport.beforeProcessingReceivedMessage = nil
+                                        partialDone()
+                                    }
+                                }
+
+                                // Before the sync ends
+                                channel.presenceMap.testSuite_injectIntoMethodBefore(#selector(ARTPresenceMap.endSync)) {
+                                    expect(channel.presenceMap.members).to(haveCount(4))
+                                    expect(channel.presenceMap.localMembers).to(haveCount(1))
```

Comment:
I am not sure this demonstrates:

> . The members re-entered automatically must be removed from the internal PresenceMap ensuring that members present on the channel are constructed from presence events sent from Ably since the channel became ATTACHED

Can you check that when the `Enter` event is sent to Ably that the member has been removed from `localMembers` and is only later added back in when the lib receives the ACK please?